### PR TITLE
qdiskinfo: init at 0.2

### PR DESCRIPTION
**Pull request by RoyDubnium**
## Description of changes

A native linux alternative to CrystalDiskInfo. Provides SMART information about ATA and NVMe drives.
https://github.com/edisionnano/QDiskInfo

Resubmit of #325304
## Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [X] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/qd/qdiskinfo/package.nix [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    hash = "sha256-YLNpQOmh/exBJDuYRUEvzNv/El4x+PttEoj+aCur9wg=";
+  };
+
+  nativeBuildInputs = [
+    cmake
+    qt6.wrapQtAppsHook
+  ];
+
+  buildInputs = [
+    qt6.full
```
Comment (AndersonTorres) [inline pkgs/by-name/qd/qdiskinfo/package.nix:27]: is this needed? Isn't there a small set of subpackages?

Comment (RoyDubnium) [reply pkgs/by-name/qd/qdiskinfo/package.nix]: It would appear so. The derivation builds with just qtbase.

**Comment by donovanglover** [comment]
Add `qtwayland` to buildInputs for Wayland support

**Comment by RoyDubnium** [comment]
Added

File: pkgs/by-name/qd/qdiskinfo/package.nix [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+  ];
+
+  cmakeFlags = [
+    "-DCMAKE_BUILD_TYPE:STRING=MinSizeRel"
+    "-DQT_VERSION_MAJOR=6"
+  ];
+
+  postInstall = ''
+    wrapProgram $out/bin/QDiskInfo \
+      --prefix PATH : ${smartmontools}/bin
```
Comment (Atemu) [inline pkgs/by-name/qd/qdiskinfo/package.nix:39]: Does it expect to see the same version of smartmontools that it saw at build time? If not, this should be suffixed in order to not violate the principle of least surprise.

```suggestion
      --suffix PATH : ${smartmontools}/bin
```

**Review by Atemu — APPROVED** [review]
Diff LGTM and runs fine for me.

Result of `nixpkgs-review pr 334388` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>qdiskinfo</li>
  </ul>
</details>

**Review by Atemu — APPROVED** [review]
Result of `nixpkgs-review pr 334388` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>qdiskinfo</li>
  </ul>
</details>

**Pull request merged**